<launch>
	<!-- Config file with initial velocities for each robot. -->
	<arg name="filename" default="$(find sphero_formation)/cfg/sphero_init_vel.cfg"/>

	<!-- Start Reynolds controller nodes launcher. -->
	<node pkg="sphero_formation" type="reynolds_launch.sh" name="reynolds_launcher" args="$(arg filename) robot" output="screen"/>

	<!-- Start nearest_search node: search for other robots in range. -->
	<node pkg="sphero_formation" type="nearest_search.py" name="search" output="screen">
		<param name="robot_name" type="string" value="robot"/>
	</node>
	

	<!-- Record a bag for analysis purposes -->
	<arg name="bag_name"/>
	<arg name="rosbag_args" default='-O $(find sphero_formation)/data/$(arg bag_name).bag -a'/>
	<node pkg="rosbag" type="record" name="rosbag_record" args="$(arg rosbag_args)" output="screen"/>
</launch>
